src/main/kotlin/car/domain/Cars.kt

```diff
@@ -0,0 +1,31 @@
+package car.domain
+
+class Cars(private val cars: List<Car>, private val energyProvider: EnergyProvider = RandomEnergyProvider()) {
+
+    private val _histories: MutableList<CarMoveHistoryCollection> = ArrayList()
```

Comment:
코틀린은 읽기 전용 컬렉션과 변경 가능한 컬렉션을 구별해 제공합니다. 아래의 코드로 개선해 보면 어떨까요?

```suggestion
    private val _histories: MutableList<CarMoveHistoryCollection> = mutableListOf()
```